test/lsp_test_runner.cc

```diff
@@ -425,10 +425,15 @@ TEST_CASE("LSPTest") {
                 if (auto defAssertion = dynamic_pointer_cast<DefAssertion>(assertion)) {
                     version = defAssertion->version;
                     symbol = defAssertion->symbol;
+                    // Some definition locations are not the definition of themselves.
+                    if (!defAssertion->isDefOfSelf) {
+                        continue;
+                    }
```

Comment:
This doesn't make sense to me. Can you update the README with an explanation of what it means to use `not-def-of-self`?

Reply:
Ok, I'm reading the test a little more closely below, and as I understand it, it's something like this:

- Normally, `def` / `usage` assertions come in groups, and jump to def for everything in the group, including both `def`s and `usage`s, jump to the `def` assertion.
- But for package imports, that's not the case; there's a level of indirection, where `usage`s in the package jump to the import `def`, but the import `def` actually jumps to a different `def` in the other package.

Given that, this seems *very* specific to package imports. ooc, why have you chosen to give this a seemingly generic sounding name (`not-def-of-self`)? Do you anticipate further non-package related usages? I wonder if something like a new `import-def:` assertion would make sense instead of an option that's unused by most assertion kinds.

Reply:
Your description is spot-on!

> Do you anticipate further non-package related usages?

I figured that this pattern might show up in other rewriter-like passes, and went for a general-shaped solution rather than a specific solution.